pkg/controller/certificates/sync.go

```diff
@@ -404,6 +406,8 @@ func (c *controller) updateSecret(ctx context.Context, crt *v1alpha1.Certificate
 		if err != nil {
 			return nil, fmt.Errorf("invalid certificate data: %v", err)
 		}
+	case utilfeature.DefaultFeatureGate.Enabled(feature.DisableTmpCrt):
```

Comment:
Once you invert this to be `EnableTemporaryCertificate`, you'll also need to flip this to check `!utilfeature.DefaultFeatureGate.Enabled` (i.e. check if it *isn't* enabled)